File: 프로그래머스/1/161990. 바탕화면 정리/바탕화면 정리.js
function solution(wallpaper) {
    return findSquare(wallpaper);
}
    
const findSquare = (wallpaper) => {
    const result = [wallpaper.length, wallpaper[0].length, 0, 0];
    
    for (let i = 0; i < wallpaper.length; i++) {
        for (let j = 0; j < wallpaper[0].length; j++) {
            const now = wallpaper[i][j];
            if (now !== '#') {
                continue;
            }
            if (i < result[0]) {
                result[0] = i;
            }
            if (j < result[1]) {
                result[1] = j;
            }
            if (i >= result[2]) {
                result[2] = i + 1;
            }
            if (j >= result[3]) {
                result[3] = j + 1;
            }
        }
    }

    return result;
}

File: 프로그래머스/1/161990. 바탕화면 정리/바탕화면 정리.java
class Solution {
    public int[] solution(String[] wallpaper) {
        int[] result = {49, 49, 1, 1};
        
        for (int i = 0; i < wallpaper.length; i++) {
            String line = wallpaper[i];
            for (int j = 0; j < line.length(); j++) {
                char point = line.charAt(j);
                if (point == '.') {
                    continue;
                }
                updateResult(result, i, j);
            }
        }
        
        return result;
    }
    
    private void updateResult(int[] result, int row, int col) {
        if (result[0] > row) {
            result[0] = row;
        }
        if (result[1] > col) {
            result[1] = col;
        }
        if (result[2] - 1 < row) {
            result[2] = row + 1;
        }
        if (result[3] - 1 < col) {
            result[3] = col + 1;
        }
        
    }
}